### PR TITLE
IGNITE-12941 .NET: Fix and document single file deployment on .NET 5

### PR DESCRIPTION
**Pull request by ptupitsyn**
* Fix `DllNotFoundException: Unable to load shared library 'libcoreclr.so'` error in single file mode on .NET 5 by adding a custom `DllImportResolver`
* Update Deployment and Troubleshooting documentation

### CODE REVIEW

File: modules/platforms/dotnet/Apache.Ignite.Core/Impl/Unmanaged/NativeLibraryUtils.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            }
+
+            var resolveMethod = typeof(NativeLibraryUtils).GetMethod("Resolve", BindingFlags.Static | BindingFlags.NonPublic);
+            Debug.Assert(resolveMethod != null);
+
+            var libraryName = Expression.Parameter(typeof(string));
+            var assembly = Expression.Parameter(typeof(Assembly));
+            var searchPath = Expression.Parameter(typeof(Nullable<>).MakeGenericType(dllImportSearchPathType));
+
+            var resolve = Expression.Call(resolveMethod, libraryName);
```
Comment (gurustron) [inline modules/platforms/dotnet/Apache.Ignite.Core/Impl/Unmanaged/NativeLibraryUtils.cs:73]: Maybe change `resolve` to :

```
  Expression<Func<string, IntPtr>> call = s => Resolve(s);
  var resolve = Expression.Invoke(call, libraryName);
```

I think it is a little bit cleaner.

**Review by isapego — APPROVED** [review]

File: modules/platforms/dotnet/Apache.Ignite.Core/Impl/Unmanaged/NativeLibraryUtils.cs (hunk 1 above)
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite.Core/Impl/Unmanaged/NativeLibraryUtils.cs]: Good point, fixed

File: docs/_docs/net-specific/net-deployment-options.adoc [hunk 2]
```diff
@@ -52,6 +52,17 @@ tab:MyApp.csproj[]
 ----
 --
 
+== Single File Deployment
+
+Ignite.NET supports https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file[single file deployment] that is available in .NET Core 3 / .NET 5+.
```
Comment (Nikita-tech-writer) [inline docs/_docs/net-specific/net-deployment-options.adoc:57]:
```suggestion
Ignite.NET supports link:https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file[single file deployment] that is available in .NET Core 3 / .NET 5+.
```

File: docs/_docs/net-specific/net-deployment-options.adoc [hunk 3]
```diff
@@ -52,6 +52,17 @@ tab:MyApp.csproj[]
...
+
+* Use `IncludeAllContentForSelfExtract` MSBuild property to include jar files into the single-file bundle, or ship them separately.
```
Comment (Nikita-tech-writer) [inline docs/_docs/net-specific/net-deployment-options.adoc:59]:
```suggestion
* Use the `IncludeAllContentForSelfExtract` MSBuild property to include jar files into the single-file bundle, or ship them separately.
```

**Review by Nikita-tech-writer — COMMENTED** [review]
LGTM. Just a couple of minor suggeted changes.

**Pull request merged**